packages/vscode-extension/src/utilities/license.ts

```diff
@@ -0,0 +1,51 @@
+import fetch from "node-fetch";
+import { extensionContext } from "./extensionContext";
+import path from "path";
+import { Platform } from "./platform";
+import { exec } from "./subprocess";
+
+const TOKEN_KEY = "token_key";
+const BASE_CUSTOMER_PORTAL_URL = "https://portal.ide.swmansion.com/";
+
+export async function activateDevice(licenseKey: string, username: string) {
+  const url = new URL("/api/create-token", BASE_CUSTOMER_PORTAL_URL);
+  const body = {
+    fingerprint: await generateDeviceFingerprint(),
+    name: username,
+    licenseKey,
+  };
+
+  const response = await fetch(url, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(body),
+  });
+
+  let newToken;
+
+  if (response.ok) {
+    const responseBody = await response.json();
+    newToken = responseBody.token as string;
+    extensionContext.secrets.store(TOKEN_KEY, newToken ?? "");
+    return true;
+  }
+
+  return false;
+}
+
+async function generateDeviceFingerprint() {
+  const simControllerBinary = path.join(
+    extensionContext.extensionPath,
+    "dist",
+    Platform.select({ macos: "simulator-server-macos", windows: "simulator-server-windows.exe" })
+  );
+  const { stdout } = await exec(simControllerBinary, ["fingerprint"]);
+  return stdout;
```

Comment:
yes

Reply:
If for some reason simserver fails with exit code != 0, does this function fail immediately? We'd like to avoid setting the machine fingerprint to some error message or empty string